manim/mobject/graphing/functions.py

```diff
@@ -14,6 +14,7 @@
 from manim.mobject.graphing.scale import LinearBase, _ScaleBase
 from manim.mobject.opengl.opengl_compatibility import ConvertToOpenGL
 from manim.mobject.types.vectorized_mobject import VMobject
+from manim.typing import Point2D, Point3D
```

Comment:
I would prefer for `TYPE_CHECKING` to be imported from `typing`, and this to be added as something like
```suggestion
if TYPE_CHECKING:
    from manim.typing import Point2D, Point3D
```

Reply:
I see...why do we use the `TYPE_CHECKING` variable? (sorry, I'm still quite new to all of this 😓 )

Reply:
We use it to reduce library overhead. `TYPE_CHECKING` is only`True` at typechecking time, not runtime, so anything inside will not be imported at runtime.